server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/Rights.scala

```diff
@@ -104,15 +103,17 @@ object Rights {
     }
     .fold(EMPTY)(_ combine _)
 
-  private def toRights(entryKey: EntryKey, aclRights: Rfc4314Rights): Rights = of(Username.of(entryKey.getName), aclRights.toRights)
+  private def toRights(entryKey: EntryKey, aclRights: Rfc4314Rights): Rights = of(entryKey, aclRights.toRights)
 
   private def isSupported(key: EntryKey): Boolean = key match {
```

Comment:
While we are at it do we support negative entry key?

Reply:
that's exactly what i was gonna ask next: using `EntryKey` instead of `Username` is so that negative entry keys, group entry keys, etc are all de facto supported now.
i suggest to write some tests about it ?